contracts/BC_fusion/StakePool.sol

```diff
@@ -0,0 +1,248 @@
+        bytes32 hash = keccak256(abi.encodePacked(_delegator, _unbondSequence[_delegator]));
+
+        uint256 unlockTime = block.timestamp + IStakeHub(STAKE_HUB_ADDR).getUnbondTime();
+        UnbondRequest memory request = UnbondRequest({sharesAmount: _sharesAmount, bnbAmount: _bnbAmount, unlockTime: unlockTime});
+        _unbondRequests[hash] = request;
+        _unbondRequestsQueue[_delegator].pushBack(hash);
+
+        emit UnbondRequested(_delegator, _sharesAmount, _bnbAmount, request.unlockTime);
+        return _bnbAmount;
+    }
+
+    function redelegate(address _delegator, uint256 _sharesAmount) external onlyStakeHub whenNotPaused returns (uint256) {
+        require(_sharesAmount <= _sharesOf(_delegator), "INSUFFICIENT_BALANCE");
+
+        // calculate the BNB amount and update state
+        uint256 _bnbAmount = getPooledBNBByShares(_sharesAmount);
+        _burn(_delegator, _sharesAmount);
+        _totalPooledBNB -= _bnbAmount;
+
+        (bool success,) = STAKE_HUB_ADDR.call{value: _bnbAmount}("");
+        require(success, "TRANSFER_FAILED");
+        return _bnbAmount;
+    }
+
+    function claim(address payable _delegator, uint256 number) external onlyStakeHub whenNotPaused returns (uint256) {
+        require(_unbondRequestsQueue[_delegator].length() != 0, "NO_UNBOND_REQUEST");
+        // number == 0 means claim all
+        if (number == 0) {
+            number = _unbondRequestsQueue[_delegator].length();
+        }
+        if (number > _unbondRequestsQueue[_delegator].length()) {
+            number = _unbondRequestsQueue[_delegator].length();
+        }
+        require(number <= MAX_CLAIM_NUMBER, "TOO_MANY_REQUESTS"); // prevent too many loop in one transaction
+
+        uint256 _totalShares;
+        uint256 _totalBnbAmount;
+        while (number != 0) {
+            bytes32 hash = _unbondRequestsQueue[_delegator].front();
+            UnbondRequest memory request = _unbondRequests[hash];
+            if (block.timestamp < request.unlockTime) {
+                break;
+            }
+
+            // request is non-existed(should not happen)
+            if (request.sharesAmount == 0 && request.unlockTime == 0) {
+                continue;
+            }
+
+            _totalShares += request.sharesAmount;
+            _totalBnbAmount += request.bnbAmount;
+
+            // remove from the queue
+            _unbondRequestsQueue[_delegator].popFront();
+            delete _unbondRequests[hash];
+
+            number -= 1;
+        }
+
+        _lockedShares[_delegator] -= _totalShares;
+        (bool success,) = _delegator.call{value: _totalBnbAmount}("");
+        require(success, "CLAIM_FAILED");
+
+        emit UnbondClaimed(_delegator, _totalShares, _totalBnbAmount);
+        return _totalBnbAmount;
+    }
+
+    function distributeReward() external payable onlyStakeHub {
+        uint256 _bnbAmount = msg.value;
+        _totalReceivedReward += _bnbAmount;
```

Comment:
`_totalReceivedReward` seems useless

Reply:
Yes. It's just for statistic. I can remove it if we don't need that